src/proof/alethe/alethe_post_processor.h

```diff
@@ -99,6 +99,43 @@ class AletheProofPostprocessCallback : public ProofNodeUpdaterCallback
                            CDProof& cdp);
 };
 
+/**
+ * Final callback class used by the Alethe converter to add the last step to a
+ * proof in certain cases.
+ */
+class AletheProofPostprocessFinalCallback : public ProofNodeUpdaterCallback
+{
+ public:
+  AletheProofPostprocessFinalCallback(ProofNodeManager* pnm);
+  ~AletheProofPostprocessFinalCallback() {}
+  /** Should proof pn be updated? It should, if the last step is (cl false) or
+   * if it is an assumption.
+   */
+  bool shouldUpdate(std::shared_ptr<ProofNode> pn,
+                    const std::vector<Node>& fa,
+                    bool& continueUpdate) override;
+  /**
+   * This method gets a proof node pn = false which is printed as (cl false) and
```

Comment:
I'm not sure of the part "proof node pn = false", please rephrase.

Reply:
I rephrased it, lmk if it is still unclear. It also becomes clearer in the implementation file.